code/go/0chain.net/smartcontract/minersc/sc.go

```diff
@@ -182,70 +182,26 @@ func getHostnameAndPort(burl string) (string, int, error) {
 	return "", 0, errors.New(burl + " is not a valid url. It not a valid IP or valid DNS name")
 }
 
-func getGlobalNode(balances cstate.StateContextI) (
-	gn *GlobalNode, err error) {
-
+func getGlobalNode(
+	balances cstate.StateContextI,
+) (gn *GlobalNode, err error) {
 	gn = new(GlobalNode)
 	var p util.Serializable
 	p, err = balances.GetTrieNode(GlobalNodeKey)
-	if err != nil && err != util.ErrValueNotPresent {
-		return nil, err
-	}
-
-	if err == nil {
-		if err = gn.Decode(p.Encode()); err != nil {
-			return nil, fmt.Errorf("%w: %s", common.ErrDecoding, err)
+	if err != nil {
+		if err != util.ErrValueNotPresent {
+			return nil, err
+		}
+		gn.readConfig()
```

Comment:
Perhaps, we should return error here instead of using values from local yaml file.